### ISSUE TITLE
Add mat-table inline editing for menus/drop-down lists

### ISSUE THREAD

**Issue by yukund** [angular/material2#9552]
#### Bug, feature request, or proposal:
feature request : mat-table inline editing with menus/drop-down-lists

#### What is the expected behavior?
Clicking on a table cell should show inline menus shown in below.
![image](https://user-images.githubusercontent.com/4123840/35301989-fa577408-00f1-11e8-926c-26079d7f0b56.png) 

image is from https://material.io/guidelines/components/data-tables.html#data-tables-interaction

#### What is the current behavior?

N/A
#### What are the steps to reproduce?
N/A

#### What is the use-case or motivation for changing an existing behavior?
"users may need to select from a predefined list of options. In this scenario, embed a menu component directly in the table."

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 5 

#### Is there anything else we should know?
related issue : https://github.com/angular/material2/issues/5982

**Comment by andrewseguin** [comment]
Not sure the request here, do you want the table to have built-in support for a cell containing a menu? Have you tried just using mat-menu?

**Comment by yukund** [comment]
Hi @andrewseguin,  thank you for your reply.

> Not sure the request here, do you want the table to have built-in support for a cell containing a menu?

Yes, I want to have build-in support for a inline menu.
I am thinking if I can pass relevant parameters to the table component (or cell directive), the table component then renders an inline menu based on the provided parameters. For example,  the parameters could be the cell is editable, type is drop down menu,  a list of items should be shown in the menu. It is just my thought, please let me know if it makes science. 

> Have you tried just using mat-menu?

Not yet.  I am going to try it. Do you have any example like showing mat-menu on mat-table?  
basically,  if users click the table cell I just show a predefined mat-menu or mat-list?

**Comment by andrewseguin** [comment]
Sure here's an example of putting a menu in a table cell 

https://stackblitz.com/edit/angular-qhmrwb?file=app/table-basic-example.html

We'll eventually want to introduce easy drop-in columns like this, but not sure that menu is a use case yet (first one will likely be a simple text-based column).

For now I think just putting a menu trigger in will solve your case.

**Comment by yukund** [comment]
@andrewseguin , thank you for the example. I have a question.
you added a button to a cell and used the  button to trigger a menu 
```
  <mat-cell *matCellDef="let element"> 
        <button mat-button [matMenuTriggerFor]="menu">
          I'm a menu button
        </button> 
  </mat-cell>

<mat-menu #menu="matMenu">
  <button mat-menu-item>Item 1</button>
  <button mat-menu-item>Item 2</button>
</mat-menu>

```

If I would like to use **mat-select instead of mat-menu,** how should I trigger it when users click a table cell?  thanks
the select looks like that
```
<mat-form-field>
  <mat-select placeholder="Favorite food">
    <mat-option *ngFor="let food of foods" [value]="food.value">
      {{ food.viewValue }}
    </mat-option>
  </mat-select>
</mat-form-field>
```

**Comment by andrewseguin** [comment]
It's really no different, the select should handle itself just fine. Just put a select into the cell

https://stackblitz.com/edit/angular-qhmrwb-qyznkn?file=app/table-basic-example.html

**Comment by ahainen** [comment]
> It's really no different, the select should handle itself just fine. Just put a select into the cell
> 
> https://stackblitz.com/edit/angular-qhmrwb-qyznkn?file=app/table-basic-example.html

Is there any way to have the arrow_drop_down pin to the end of the text? To the user, without vertical lines in the data table, the arrow_drop_down looks near the column to it's right until hovered or clicked.

**Comment by mdaslamknlgit** [comment]
Hi

How to setfocus on the mat-input after selection of drop down list

thanks
